### PR TITLE
Added packaging configuration to generate a windows installer

### PR DESCRIPTION
**Pull request by ThijsSassen**
This commit adds configuration to create a proper windows installer when using the package target.
It also provides mechanisms to generate a debian package file on debian systems and an rpm package file on redhat based systems.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,11 @@
 cmake_minimum_required(VERSION 3.7)
 project(CycloneDDS-CXX VERSION 0.1.0 LANGUAGES C CXX)
 
+set(PROJECT_NAME_FULL "Eclipse Cyclone DDS ISO IEC C++ PSM")
+
+# Set some convenience variants of the project-name
+string(REPLACE " " "-" PROJECT_NAME_DASHED "${PROJECT_NAME_FULL}")
+
```
Comment (k0ekk0ek) [inline CMakeLists.txt:19]: Please set this in the CMakeCPack.cmake file. It's not used anywhere in the project, and I've just removed all those silly instances in the xtypes branch.

File: CMakeLists.txt [hunk 2]
```diff
@@ -189,3 +194,5 @@ endif()
 if(BUILD_DOCS)
   add_subdirectory(docs)
 endif()
+
+include (${CMAKE_CURRENT_SOURCE_DIR}/CPackConfig.cmake)
```
Comment (k0ekk0ek) [inline CMakeLists.txt:198]: No need to prefix with `${CMAKE_CURRENT_SOURCE_DIR}`

File: PackageConfig.cmake.in [hunk 3]
```diff
@@ -16,3 +16,4 @@ find_package(CycloneDDS REQUIRED)
 include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
 
 check_required_components("@PROJECT_NAME@")
+include("${CMAKE_CURRENT_LIST_DIR}/idlcxx/Generate.cmake")
```
Comment (k0ekk0ek) [inline PackageConfig.cmake.in:19]: Before `idlcxx` can be merged to `master`, `Generate.cmake` will be removed. Instead, the `Generate.cmake` file in the cyclonedds repository should gain a `LANGUAGE` keyword. The reason behind it being the real fix for [issue 697 in cyclonedds](https://github.com/eclipse-cyclonedds/cyclonedds/issues/697). The logic for scanning dependencies is probably not something we want to maintain in each repository.

**Review by k0ekk0ek — CHANGES_REQUESTED** [review]
@ThijsSassen, I really like the installer. Especially the part where you can opt to add the bin directory to the path. Kudos! Had to study CPack a bit to decide if this is the right way to do it. We're almost there. In my search I stumbled upon the [PackageGenerators](https://gitlab.kitware.com/cmake/community/-/wikis/doc/cpack/PackageGenerators) page in the CMake wiki and from there it was an easy step to simply consult [CMake's own repository](https://gitlab.kitware.com/cmake/cmake/) to see what they do. The `CPackConfig.cmake` file is normally generated according to the documentation. It's also the file that the `cpack` command itself will use, so including it from `CMakeLists.txt` doesn't quite seem to be the way to go as it's not meant to be used by CMake. The Kitware guys call it `CMakeCPack.cmake` and they use a separate file named `CMakeCPackOptions.cmake.in` to set installer properties. That file will become `CMakeCPackOptions.cmake` in the build directory, which is included by `CPackConfig.cmake` by setting `CPACK_PROJECT_CONFIG_FILE`. If you can address those, I'm more than happy to pull these changes. While you're at it, can you please also change the name from `Wix` to `WiX` and remove the `CycloneDDS_` prefix from the banner etc?

File: CMakeLists.txt (hunk 1 above)
Comment (ThijsSassen) [reply CMakeLists.txt]: i moved it to CMakeCPack.cmake

File: CMakeLists.txt (hunk 2 above)
Comment (ThijsSassen) [reply CMakeLists.txt]: adjusted it

File: PackageConfig.cmake.in (hunk 3 above)
Comment (ThijsSassen) [reply PackageConfig.cmake.in:19]: I'll leave this for now as what you are suggesting requires changes in cyclonedds and given that at this moment that the idlcxx branch is replaced by the xtypes branch and that one is not yet compatible with the idlcxx branch of cyclonedds-cxx i suggest to do these changes once the xtypes branch is merged to master.

Comment (k0ekk0ek) [reply PackageConfig.cmake.in:19]: That'll work.

**Pull request merged**